configdialog.cpp

```diff
@@ -92,8 +111,23 @@ void ConfigDialog::refresh() {
 		SendMessage(slider, TBM_SETPOS, TRUE, encodef(m_current_config.at(entry.first)));
 	}
 
+	if (m_current_config.at(YonkPattern) < 0 || m_current_config.at(YonkPattern) >= _PATTERN_COUNT)
+	{
+		MessageBox(NULL, (LPCWSTR)L"Yonk Pattern was set to an out of range option!\nIt will be reset to default.", (LPCWSTR)L"Yok.scr", MB_ICONWARNING | MB_OK);
+		m_current_config.at(YonkPattern) = 0;
+	}
 	HWND pattern_box = GetDlgItem(m_dialog, IDC_YONK_PATTERN);
 	ComboBox_SelectString(pattern_box, -1, pattern_strings.at(
-		(SpritePattern) m_current_config.at(YonkPattern)).c_str()
+		(SpritePattern)m_current_config.at(YonkPattern)).c_str()
+	);
+
+	if (m_current_config.at(YonkPalette) < 0 || m_current_config.at(YonkPalette) >= static_cast<int>(PaletteGroup::_PALETTE_OPTION_COUNT))
```

Comment:
Why does there need to be a `static_cast` here, but not on the previous if? And as a style point (that VS is not set up to enforce), we should use C style casts instead of `static_cast`.

Reply:
Wasn't aware that a C-style cast works here. And it's needed because the compiler doesn't like comparing an int to an enum class member.